### PR TITLE
chore: transfer sequence example

### PR DESCRIPTION
**Pull request by harrysolovay**
An example that showcases sequential txs

### CODE REVIEW

File: examples/transfer_sequence.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { alice, bob, charlie, dave, Rune, RunicArgs, Sr25519, ValueRune } from "capi"
+import { Balances, System } from "westend_dev/mod.ts"
+import { upperFirstCase } from "../deps/case.ts"
+
+await aliceFree()
+  .log(msg("Alice", "before"))
+  .chain(() =>
+    Object
+      .entries({ bob, charlie, dave })
+      .reduce<ValueRune<any, any>>((acc, args) =>
```
Comment (harrysolovay) [inline examples/transfer_sequence.ts:10]: This isn't the prettiest (both the `reduce` call, and the `ValueRune` type args. How shall we clean this up? Is a new Rune utility required?

File: examples/transfer_sequence.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  return Rune.str`${name} balance ${stage}:`
+}
+
+function aliceFree() {
+  return System.Account.entry([alice.publicKey]).access("data", "free")
+}
+
+function beforeThenTxThenAfter<X>(...args: RunicArgs<X, [name: string, pair: Sr25519]>) {
+  const pair = Rune.resolve(args[1])
+  const name = Rune.resolve(args[0]).map(upperFirstCase)
```
Comment (harrysolovay) [inline examples/transfer_sequence.ts:28]: #589

**Comment by ryanleecode** [comment]
Why do it this way instead of via batch call?

**Comment by harrysolovay** [comment]
> Why do it this way instead of via `batchAll` call?

Because the goal here is to show off modeling sequential txs with Capi's Rune system (not making use of the `Utility` pallet). Could be worth adding a designated `batchAll` example as well though.

**Comment by ryanleecode** [comment]
```ts
import { alice, bob, charlie, dave, Rune, RunicArgs, Sr25519, ValueRune } from "capi"
import { Balances, System } from "westend_dev/mod.ts"
import { upperFirstCase } from "../deps/case.ts"

await getBalance(alice.publicKey)
  .log(msg("Alice", "before"))
  .chain(() =>
    Object
      .entries({ bob, charlie, dave })
      .reduce<ValueRune<any, any>>((acc, args) =>
        acc
          .chain(() => transfer(...args))
          .into(ValueRune), Rune.constant(undefined))
  )
  .chain(() => getBalance(alice.publicKey).log(msg("Alice", "after")))
  .run()

function msg<X>(...[name, stage]: RunicArgs<X, [name: string, stage: "before" | "after"]>) {
  return Rune.str`${name} balance ${stage}:`
}

function transfer<X>(...args: RunicArgs<X, [name: string, pair: Sr25519]>) {
  const pair = Rune.resolve(args[1])
  const name = Rune.resolve(args[0]).map(upperFirstCase)
  return getBalance(pair.access("publicKey"))
    .log(msg(name, "before"))
    .chain(() =>
      Balances
        .transfer({
          dest: pair.access("address"),
          value: 1000123n,
        })
        .signed({ sender: alice })
        .sent()
        .logStatus(Rune.str`Transfer to ${name} from alice:`)
        .finalized()
    )
    .chain(() => getBalance(pair.access("publicKey")).log(msg(name, "after")))
}

function getBalance<X>(...[pubKey]: RunicArgs<X, [pubKey: Uint8Array]>) {
  return System.Account.entry(Rune.tuple([pubKey])).access("data", "free")
}
```

Rewrote it a little and change the some naming to be more obvious

**Comment by harrysolovay** [comment]
The only different I notice is that you've parameterized the public key of the balance read... not sure if that's necessary for this example. Less code / more legible if the public key is supplied directly... if I missed any other changes, want to suggest them as commits within the GH review UI?

**Review by tjjfvi — APPROVED** [review]

**Pull request merged**